### PR TITLE
doc/md/guides: automatic migration planning for gorm

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: doc/md/guides/orms/gorm.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
+---
+id: gorm
+title: Automatic migration planning for GORM
+slug: /guides/orms/gorm
+---
+
+## TL;DR
+* [GORM](https://gorm.io) is a an ORM library that's widely used in the Go community.
```
Comment (hilakashai) [inline doc/md/guides/orms/gorm.md:8]:
```suggestion
* [GORM](https://gorm.io) is an ORM library that's widely used in the Go community.
```

File: doc/md/guides/orms/gorm.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+The desired state can be provided to Atlas via an Atlas schema [HCL file](https://atlasgo.io/atlas-schema/sql-resources) 
+or as a connection string to a database that contains the desired schema.
+
+In this guide, we will show how Atlas can automatically plan schema migrations for
+GORM users.
+
+## Prerequisites
+
+* An existing project with a GORM.
```
Comment (hilakashai) [inline doc/md/guides/orms/gorm.md:41]:
```suggestion
* An existing GORM project
```

File: doc/md/guides/orms/gorm.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+* Developers using GORM can use Atlas to automatically plan schema migrations
+  for them, based on the desired state of their schema instead of crafting them by hand.
+
+## Automatic migration planning for GORM
+
+GORM is a popular ORM widely used in the Go community. GORM allows users to 
+manage their database schemas using its [AutoMigrate](https://gorm.io/docs/migration.html#Auto-Migration)
+feature, which is usually suffucient during development and in many simple cases. 
+
+However, at some point, many teams need more control and decide to employ
```
Comment (a8m) [inline doc/md/guides/orms/gorm.md:20]:
```suggestion
However, at some point, teams need more control and decide to employ 
```

File: doc/md/guides/orms/gorm.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  for them, based on the desired state of their schema instead of crafting them by hand.
+
+## Automatic migration planning for GORM
+
+GORM is a popular ORM widely used in the Go community. GORM allows users to 
+manage their database schemas using its [AutoMigrate](https://gorm.io/docs/migration.html#Auto-Migration)
+feature, which is usually suffucient during development and in many simple cases. 
+
+However, at some point, many teams need more control and decide to employ 
+a [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations)
```
Comment (a8m) [inline doc/md/guides/orms/gorm.md:21]:
```suggestion
the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) 
```

File: doc/md/guides/orms/gorm.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+feature, which is usually suffucient during development and in many simple cases. 
+
+However, at some point, many teams need more control and decide to employ 
+a [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) 
+methodology. Once this happens, the responsibility for planning migration scripts and making
+sure they are in line with what GORM expects at runtime is moved to developers.
+
+Atlas can automatically plan database schema migrations for developers using GORM.
+Atlas plans migrations by calculating the diff between the _current_ state of the database,
+and it's _desired_ state.
```
Comment (a8m) [inline doc/md/guides/orms/gorm.md:27]:
```suggestion
and its _desired_ state.
```

File: doc/md/guides/orms/gorm.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+```
+
+As reference for the next steps, the URL for the Dev Database will be:
+```text
+mysql://root:pass@localhost:3306/dev
+```
+
+## Create an auto-migration script
+
+Depends on how your project is set up, you may be able to skip this part.
```
Comment (hilakashai) [inline doc/md/guides/orms/gorm.md:63]: Maybe add some context? Who can skip this part? Is there some more specific info?

File: doc/md/guides/orms/gorm.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+		panic(err)
+	}
+}
+
+func init() {
+	flag.StringVar(&conn, "conn", "", "connection string to db")
+}
+```
+
+Create a schema named `gorm` in our dev database to hold the desired state:
```
Comment (hilakashai) [inline doc/md/guides/orms/gorm.md:114]:
```suggestion
Create a schema named `gorm` in our Dev Database to hold the desired state:
```

File: doc/md/guides/orms/gorm.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+in the `migrations` directory (currently an empty schema), to the desired schema
+which exists in the Dev Database.
+
+Run:
+
+```text
+atlas migrate diff --dir file://migrations --dev-url mysql://root:pass@localhost:3306/dev --to mysql://root:pass@localhost:3306/gorm
+```
+
+Observe two new files were created under the `migrations` directory:
```
Comment (hilakashai) [inline doc/md/guides/orms/gorm.md:143]:
```suggestion
Observe that two new files were created under the `migrations` directory:
```

File: doc/md/guides/orms/gorm.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+	}
+	// Replace `&Product{}, &User{}` with the models of your application.
+	if err := db.AutoMigrate(&Product{}, &User{}); err != nil {
+		panic(err)
+	}
+}
+
+func init() {
+	flag.StringVar(&conn, "conn", "", "connection string to db")
+}
```
Comment (a8m) [inline doc/md/guides/orms/gorm.md:111]:
```suggestion
func main() {
        conn := flag.StringVar("conn", "", "connection string to db")
	flag.Parse()
	if *conn == "" {
		log.Fatalln("conn flag required")
	}
	db, err := gorm.Open(mysql.Open(conn), &gorm.Config{})
	if err != nil {
		log.Fatalln(err)
	}
	// Replace `&Product{}, &User{}` with the models of your application.
	if err := db.AutoMigrate(&Product{}, &User{}); err != nil {
		log.Fatalln(err)
	}
}
```

**Review by hilakashai — COMMENTED** [review]
Love it!! Quick and easy :) 
See minor comments

File: doc/md/guides/orms/gorm.md [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+go run main.go -conn 'root:pass@tcp(localhost:3306)/gorm'
+```
+
+## Use Atlas to plan an initial migration
+
+Next let's create a directory to hold our migration scripts:
+
+```text
+mkdir migrations
+```
```
Comment (a8m) [inline doc/md/guides/orms/gorm.md:130]: I think it's not necessary

File: doc/md/guides/orms/gorm.md [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+
+We can now use Atlas's `migrate diff` command to calculate a migration from the 
+current state, which can be thought of as the sum of all the migration scripts
+in the `migrations` directory (currently an empty schema), to the desired schema
+which exists in the Dev Database.
+
+Run:
+
+```text
+atlas migrate diff --dir file://migrations --dev-url mysql://root:pass@localhost:3306/dev --to mysql://root:pass@localhost:3306/gorm
```
Comment (a8m) [inline doc/md/guides/orms/gorm.md:140]:
```suggestion
atlas migrate diff --dir file://migrations --dev-url mysql://root:pass@:3306/dev --to mysql://root:pass@:3306/gorm
```
optional

**Review by a8m — APPROVED** [review]

File: doc/md/guides/orms/gorm.md (hunk 2 above)
Comment (rotemtam) [reply doc/md/guides/orms/gorm.md:41]: IMO, the project isn't a GORM project, its a project that uses GORM

File: doc/md/guides/orms/gorm.md (hunk 6 above)
Comment (rotemtam) [reply doc/md/guides/orms/gorm.md:63]: i'll rm

**Pull request merged**